lib/rex/socket.rb

```diff
@@ -258,6 +258,36 @@ def self.gethostbyname(host)
     @@resolver ? self.rex_gethostbyname(host) : ::Socket.gethostbyname(host)
   end
 
+  #
+  # Wrapper for Resolv::DNS.getresources which normalizes the return value to a
+  # list of hostnames regardless of the resource class.
+  #
+  # @param name [String] The name to lookup.
+  # @param typeclass [Symbol] The resource class to lookup, e.g. CNAME, MX, etc.
+  # @raises ArgumentError An argument error is raised when the typeclass is invalid.
+  # @return [Array<String>] The hostnames that were returned by the query.
+  def self.getresources(name, typeclass)
+    return self.rex_getresources(name, typeclass) if @@resolver
+
+    typeclass = typeclass.upcase
+    attribute = {
```

Comment:
semantic nit: This seems more apt as a constant

Reply:
With using `private_constant` too sounds good :+1:

https://stackoverflow.com/questions/2873903/how-to-i-make-private-class-constants-in-ruby